### PR TITLE
fr/response parsing

### PR DESCRIPTION
**Pull request by sbasan**
`vManageSession` http methods: `request, get, post, put, delete` now returns a `vManageResponse` object (which extends `Response` object from requests library)

`vManageResponse` methods takes advantage of recently introduced DataSequece to simplify response parsing:
```python
from vmngclient.dataclasses import Device
session = create_vManageSession(url=..., username=..., password=...)
response = session.get("/dataservice/device")
devices = response.dataseq(Device)
```
Or even better example:
```python
vbonds = session.get("/dataservice/device").dataseq(Device).filter(personality=Personality.VBOND)
len(vbonds)
```
You can also access parsed json fields below without causing JSONParseError
```python
response.payload.data # contents of { "data": ...  } defaults to None
response.payload.error # contents of { "error": ...  } defaults to None
response.payload.headers # contents of { "headers": ...  } defaults to None
```

### CODE REVIEW

File: vmngclient/response.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        except JSONDecodeError:
+            self.payload = JsonPayload()
+
+    def info(self, history: bool = False) -> str:
+        """Returns human readable string containing Request-Response contents"""
+        if history:
+            return response_history_debug(self, None)
+        return response_debug(self, None)
+
+    def parse(self, cls: Type[T]) -> T:
```
Comment (igosoft) [inline vmngclient/response.py:109]:
```suggestion
    def parse_dict(self, cls: Type[T]) -> T:
```
I'm just checking if we can have either one method or two methods following the intention of responsibility of each method . If possible I would to choose one method and always use DataSequence :)

**Review by igosoft — COMMENTED** [review]
LGTM - just one question / suggestion

File: vmngclient/response.py (hunk 1 above)
Comment (igosoft) [inline vmngclient/response.py:109]:
```suggestion
    def as_list(self, cls: Type[T]) -> T:
```

File: vmngclient/response.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        try:
+            return create_dataclass(cls, self.__get_json_data_as_dict())
+        except TypeError as e:
+            raise vManageResponseException(f"Cannot parse {cls}: {str(e)}", self)
+
+    def parse_list(self, cls: Type[T]) -> DataSequence[T]:
```
Comment (igosoft) [inline vmngclient/response.py:115]:
```suggestion
    def as_list(self, cls: Type[T]) -> DataSequence[T]:
```

**Comment by igosoft** [comment]
```python
from vmngclient.dataclasses import Device
session = create_vManageSession(url=..., username=..., password=...)
devices = session.get("/dataservice/device").as_list(Device) # for list of objects in json payload
info = session.get("/dataservice/route/to/info").as_dict(InfoClass) # for single object in json payload
```
just an idea

File: vmngclient/response.py (hunk 1 above)
Comment (sbasan) [reply vmngclient/response.py:109]: Not sure if as_dict is meaningful here. We are parsing to custom object T (internally from python dict, which was originally JSON Object).

Comment (sbasan) [reply vmngclient/response.py:109]: I was considering multiple options here including single parse method for both payload types.
I picked solution with explicit return types (thus 2 methods are required)
But im open to discussion - I think it is important to get it right now (as well as naming)

**Review by tomaszwarda — APPROVED** [review]

**Comment by tehAgitto** [comment]
@igosoft are you okay with these changes?

**Review by tomaszwarda — APPROVED** [review]

**Review by tzietkowski — APPROVED** [review]

**Pull request merged**